### PR TITLE
On Linux store rd-server.json and credential-server.json in apphome.

### LINKED ISSUE #5419: paths_linux.go: `AppHome` should be `filepath.Join(dataHome, appName)`
### Actual Behavior

it's too confusing with `AppHome:  filepath.Join(configHome, appName),`

On macOS, we're correct putting `settings.json` in `~/Library/Preferences/rancher-desktop`

On linux, `credential-server.json` and `rd-engine.json` should go in `~/.local/share/rancher-desktop`

and the factory-reset should delete them from the old location (which will happen if `rdctl factory-reset` deletes all of `~/.config/rancher-desktop`

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #5419

And 'apphome' is ~/.local/share/rancher-desktop

### CODE REVIEW

File: pkg/rancher-desktop/config/settingsImpl.ts [hunk 1]
```diff
@@ -395,6 +395,17 @@ const updateTable: Record<number, (settings: any) => void> = {
...
+  9: (_) => {
+    // On linux, this version deletes old instances of credential-server.json and rd-engine.json from paths.config
+    // These should always be in paths.appHome as they aren't config files
+    if (process.platform === 'linux') {
+      for (const filename of ['credential-server.json', 'rd-engine.json']) {
+        // Ignore nonexistent files (but if we're moving from settings 9 to 10 on linux, they will exist
+        // if the last session running settings version 9 wasn't followed by a factory-reset)
+        fs.rmSync(join(paths.config, filename), { force: true });
+      }
+    }
```
Comment (jandubois) [inline pkg/rancher-desktop/config/settingsImpl.ts:407]: Conceptually it seems just wrong to me that a schema migration can have side-effects that modify the filesystem.

A schema migration should just transform a JSON object into another JSON object and not have any observable side-effect.

E.g. the migration code should be called on `settings` API requests and deployment profiles as well (see #4939), and having those potentially touch the filesystem is just wrong.

**Review by jandubois — CHANGES_REQUESTED** [review]
I haven't reviewed the rest of the PR, but I wanted to register my strong objection to deleting files as a side effect of a settings schema migration.

The old files can simply be left behind. They should be deleted by a factory-reset, but otherwise they are not getting in the way, and are of trivial size, so just leave them alone!

File: pkg/rancher-desktop/config/settingsImpl.ts (hunk 1 above)
Comment (ericpromislow) [reply pkg/rancher-desktop/config/settingsImpl.ts]: I finally reverted that code -- thought I did that last week.

File: pkg/rancher-desktop/utils/__tests__/paths.spec.ts [hunk 2]
```diff
@@ -134,7 +134,7 @@ describe('paths', () => {
     const platform = os.platform();
 
     if (['darwin', 'linux'].includes(platform)) {
-      expect(pathsToDelete.some( dir => paths.lima.startsWith(dir))).toEqual(platform === 'darwin');
+      expect(pathsToDelete.some( dir => paths.lima.startsWith(dir))).toBeTruthy();
```
Comment (jandubois) [inline pkg/rancher-desktop/utils/__tests__/paths.spec.ts:137]: Not sure about the point of this test. Isn't is effectively verifying that

```
paths.lima === filepath.Join(paths.appHome, "lima")
```

Not sure why we need to check all the other directories. Not even sure if we need this test at all.

File: src/go/rdctl/pkg/config/config.go [hunk 3]
```diff
@@ -60,9 +60,20 @@ var (
...
+			}
+		} else {
+			configDir = os.Getenv("XDG_DATA_HOME")
+			if configDir == "" {
+				configDir = os.Getenv("HOME")
+				if configDir == "" {
+					log.Fatalf("HOME isn't defined on this linux system.")
+				}
+				configDir = filepath.Join(configDir, ".local", "share")
+			}
```
Comment (jandubois) [inline src/go/rdctl/pkg/config/config.go:76]: Not sure why `configDir` is a global variable; it seems be only used inside this function.

But I'm also confused that it is called `configDir` and not `dataDir` or `appDataDir` to make it clearer that it is no longer pointing to `os.UserConfigDir()`.

But that opens up the question if `--config-path` would have to change too, which I don't think is necessary.

File: pkg/rancher-desktop/utils/__tests__/paths.spec.ts (hunk 2 above)
Comment (ericpromislow) [reply pkg/rancher-desktop/utils/__tests__/paths.spec.ts:137]: Yeah, we no longer need this test now that `paths.appHome` means the same thing on linux and macOS.

File: src/go/rdctl/pkg/config/config.go (hunk 3 above)
Comment (ericpromislow) [reply src/go/rdctl/pkg/config/config.go:76]: Question 1:
The docstring for `DefineGlobalFlags`:
```
// DefineGlobalFlags sets up the global flags, available for all sub-commands
```

`configDir` points to the location of `rd-engine.json`. Users can override it with the `--config-path` option.

I think some of the confusion in this whole appHome/DataHome/config area is from the definition of `os.UserConfigDir()` in the Go world. On Linux it points to `${XDG_CONFIG_HOME:-$HOME/.config}` but on macOS it points to `$HOME/Library/Application Support`

File: bats/tests/containers/factory-reset.bats [hunk 4]
```diff
@@ -159,17 +159,15 @@ check_directories() {
     delete_dir=("$PATH_CONFIG")
     if is_unix; then
         delete_dir+=("$HOME/.rd")
+        delete_dir+=("$PATH_APP_HOME/credential-server.json" "$PATH_APP_HOME/lima" "$PATH_APP_HOME/rd-engine.json" "$PATH_LOGS")
```
Comment (adamkpickering) [inline bats/tests/containers/factory-reset.bats:162]: FWIW, we can change `"$PATH_APP_HOME/lima` to `"$LIMA_HOME"`.

Also, unrelated: what about `PATH_EXTENSIONS`?

**Review by adamkpickering — CHANGES_REQUESTED** [review]

File: src/go/rdctl/pkg/factoryreset/delete_data_linux.go [hunk 5]
```diff
@@ -31,7 +31,7 @@ func DeleteData(paths paths.Paths, removeKubernetesCache bool) error {
 	configPath, err := os.UserConfigDir()
 	if err != nil {
 		logrus.Errorf("Error getting config directory: %s", err)
-	} else {
+	} else if configPath != paths.Config {
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply src/go/rdctl/pkg/factoryreset/delete_data_linux.go:34]: My mistake -- I thought `paths.Config` didn't have the `rancher-desktop` part. Note that this is for Electron-specific files, which are stored in `~/.config/Rancher Desktop`. Capital letters and a space.

File: bats/tests/containers/factory-reset.bats (hunk 4 above)
Comment (ericpromislow) [reply bats/tests/containers/factory-reset.bats:162]: `PATH_EXTENSIONS` deleting needs to go into a separate issue. I'm already stretching this a bit for PRs, like replacing calls to `os.UserConfigDir()` with our own `paths` module (which is def better).

**Review by jandubois — DISMISSED** [review]
Looks good to me, but I only read the code; I did not run it.

There is one bug that needs to be fixed: `$LIMA_HOME/lima` → `$LIMA_HOME`.

File: bats/tests/containers/factory-reset.bats (hunk 4 above)
Comment (jandubois) [reply bats/tests/containers/factory-reset.bats]: Can you create an issue for dealing with `$PATH_EXTENSIONS`?

Comment (ericpromislow) [reply bats/tests/containers/factory-reset.bats]: Done in #5599

**Review by jandubois — APPROVED** [review]
Thanks, LGTM.

Note that I only reviewed the code, but didn't test the PR on Linux. Merging should be done after @adamkpickering approves it.

**Comment by jandubois** [comment]
Not sure why this PR was moved back to "draft".

Afaict all requested changes from me and @adamkpickering have been applied.

**Comment by ericpromislow** [comment]
I moved it to draft because I thought you wanted the windows APPDATA => LOCALAPPDATA change in first. They're related, but independent of one another.

**Comment by ericpromislow** [comment]
Latest change just removed the import-cycle from windows-specific Go code.

File: src/go/rdctl/pkg/factoryreset/delete_data.go [hunk 6]
```diff
@@ -100,7 +100,8 @@ func deleteUnixLikeData(paths paths.Paths, pathList []string) error {
 }
 
 func deleteLimaVM() error {
-	if err := directories.SetupLimaHome(); err != nil {
+	paths, err := p.GetPaths()
```
Comment (adamkpickering) [inline src/go/rdctl/pkg/factoryreset/delete_data.go:103]: We should check `err` here.

**Review by adamkpickering — DISMISSED** [review]
Only one comment really, and I don't think it is imperative to address it.

I did some basic testing on Linux and things seem to work as they should.

File: src/go/rdctl/pkg/config/config.go [hunk 7]
```diff
@@ -51,26 +52,28 @@ type ConnectionInfo struct {
...
+	var configDir string
 	var err error
 	if runtime.GOOS == "linux" && isWSLDistro() {
 		if configDir, err = wslifyConfigDir(); err != nil {
 			log.Fatalf("Can't get WSL config-dir: %v", err)
 		}
+		configDir = filepath.Join(configDir, "rancher-desktop")
 	} else {
-		configDir, err = os.UserConfigDir()
+		appPaths, err := paths.GetPaths()
```
Comment (jandubois) [inline src/go/rdctl/pkg/config/config.go:70]: In `cmd/shell.go` you are importing `paths.go` as `p` and use `paths` as the local variable. Here you import it by the normal name and use `appPaths` for the variable instead.

I think this should be handled consistently. I'm not a fan of using `p` for the import, so using `appPaths` seems better to me.

I would even go ahead and change `paths.GetPaths()` to `paths.GetAppPaths()` to match, but maybe that is too much for this PR?

**Review by adamkpickering — DISMISSED** [review]

**Review by adamkpickering — DISMISSED** [review]

**Review by adamkpickering — APPROVED** [review]

**Review by jandubois — APPROVED** [review]
I have various stylistics concerns, but I want the functionality merged to unblock other PRs. I'm approving under the assumptions that the concerns in #5629 are going to be addressed in a follow-up PR.

**Pull request merged**